tests/acceptance/features/apiEncryption/encoding.feature

```diff
@@ -0,0 +1,81 @@
+    Then the content of file "/textfile1.txt" for user "Alice" should be "some data"
+
+  Scenario: Download encrypted files with different encoding
+    Given the administrator has added system config key "encryption.use_legacy_encoding" with value "true" and type "boolean"
+    And user "Alice" has uploaded file with content "some data" to "/textfile1.txt"
+    And the administrator has added system config key "encryption.use_legacy_encoding" with value "false" and type "boolean"
+    And user "Alice" has uploaded file with content "some new data" to "/textfile2.txt"
+    Then the content of file "/textfile1.txt" for user "Alice" should be "some data"
+    And the content of file "/textfile2.txt" for user "Alice" should be "some new data"
+
+  Scenario: overwrite file with legacy encoding
+    Given the administrator has added system config key "encryption.use_legacy_encoding" with value "true" and type "boolean"
+    And user "Alice" has uploaded file with content "some data" to "/textfile1.txt"
+    And the administrator has added system config key "encryption.use_legacy_encoding" with value "false" and type "boolean"
+    And user "Alice" has uploaded file with content "some new data" to "/textfile1.txt"
+    Then the content of file "/textfile1.txt" for user "Alice" should be "some new data"
+
+  Scenario: overwrite file with default encoding by legacy encoding
+    Given the administrator has added system config key "encryption.use_legacy_encoding" with value "false" and type "boolean"
+    And user "Alice" has uploaded file with content "some data" to "/textfile1.txt"
+    And the administrator has added system config key "encryption.use_legacy_encoding" with value "true" and type "boolean"
+    And user "Alice" has uploaded file with content "some new data" to "/textfile1.txt"
+    Then the content of file "/textfile1.txt" for user "Alice" should be "some new data"
+
+  Scenario: share file with legacy encoding
+    Given the administrator has added system config key "encryption.use_legacy_encoding" with value "true" and type "boolean"
+    And user "Alice" has uploaded file with content "some data" to "/textfile1.txt"
+    And user "Brian" has been created with default attributes and without skeleton files
+    And user "Alice" has shared file "textfile1.txt" with user "Brian"
+    Then the content of file "/textfile1.txt" for user "Brian" should be "some data"
+
+  Scenario: share file with default encoding
+    Given the administrator has added system config key "encryption.use_legacy_encoding" with value "false" and type "boolean"
+    And user "Alice" has uploaded file with content "some data" to "/textfile1.txt"
+    And user "Brian" has been created with default attributes and without skeleton files
+    And user "Alice" has shared file "textfile1.txt" with user "Brian"
+    Then the content of file "/textfile1.txt" for user "Brian" should be "some data"
```

Comment:
There could also be a scenario where the file is uploaded and shared when legacy encoding is in effect. Then switch to new encoding and check that Brian can still download the file. And a check that Brian can overwrite the file and Alice can see the changed content.

That should be enough - to do more we would end up writing scenarios for all the combinations of group shares, public link shares, with files in folders, files in sub-folders...!